frontend/src/components/messages/NewRequestDialog.vue

```diff
@@ -493,32 +522,29 @@ export default {
         this.newRequestModel.subject = value
       },
     },
-    'newRequestModel.facilities': {
-      async handler() {
-        const isValid = await this.validateOfmProgram()
-        if (isValid) {
-          this.showFacilityNotInOFMMessage = false
-          this.isDisabled = false
-        } else if (this.isInitialLoad && this.isAnAccountMaintenanceRequest) {
-          // Only disable if this is not the initial call on watch. This is to prevent disabling the submit button when the dialog is opened for the first time.
-          this.isDisabled = true
-        }
-        this.isInitialLoad = true
-      },
-      deep: true,
-    },
     'newRequestModel.requestCategoryId': {
-      handler(value) {
+      async handler(value) {
+        const isAnIrregularExpense = value === this.getRequestCategoryIdByName(REQUEST_CATEGORY_NAMES.IRREGULAR_EXPENSES)
+        if (isAnIrregularExpense) {
+          if (!this.fundingAgreements) {
+            await this.getFundingAgreements()
+          }
+        }
+
         const isAccountMaintenance = value === this.getRequestCategoryIdByName(REQUEST_CATEGORY_NAMES.ACCOUNT_MAINTENANCE)
         this.resetModelData(isAccountMaintenance)
       },
     },
   },
-  created() {
+  async created() {
     this.PHONE_FORMAT = PHONE_FORMAT
     this.EMAIL_FORMAT = EMAIL_FORMAT
     this.AUTO_REPLY_MESSAGE = 'Your change request is complete.'
     this.setUpDefaultNewRequestModel()
+
+    if (this.PERMISSIONS.SUBMIT_CHANGE_REQUEST) {
+      this.isOFMValid = await this.validateOfmProgram()
```

Comment:
I believe the validateOfmProgram check depends on the selection of facilities which hasn't been done at this point. The best place might be on a watcher for subCategories.